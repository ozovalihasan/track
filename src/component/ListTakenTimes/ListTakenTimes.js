import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ListTakenTimes = ({
  children, takenTimes,
}) => (
  <Main>
    {
      takenTimes.map(takenTime => React.cloneElement(children, { key: takenTime.id, takenTime }))
    }

  </Main>
);

const Main = styled.div`
              background-color: blue;
              border: red 2px solid;
              display: grid;
              grid-template-columns: repeat(1, 1fr);
              grid-gap: 2px;
              overflow: scroll;
              align-items: center;
`;

ListTakenTimes.propTypes = {
  children: PropTypes.element.isRequired,
  takenTimes: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default ListTakenTimes;
