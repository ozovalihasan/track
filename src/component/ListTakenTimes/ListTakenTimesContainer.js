import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchListTakenTimes } from '../../redux';

import ListTakenTimes from './ListTakenTimes';

const ListTakenTimesContainer = ({ children }) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchListTakenTimes());
  }, []);

  const takenTimes = useSelector(state => state.takenTime.list);

  const trackedItemId = useSelector(state => state.trackedItem.chosen.trackedItem.id);

  let filteredTakenTimes;
  if (trackedItemId) {
    filteredTakenTimes = takenTimes.filter(
      takenTimes => takenTimes.tracked_item_id === trackedItemId,
    );
  } else {
    filteredTakenTimes = takenTimes;
  }

  return (
    <ListTakenTimes takenTimes={filteredTakenTimes}>
      {children}
    </ListTakenTimes>
  );
};

ListTakenTimesContainer.propTypes = {
  children: PropTypes.element.isRequired,
};
export default ListTakenTimesContainer;
