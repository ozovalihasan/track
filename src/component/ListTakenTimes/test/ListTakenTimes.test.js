import { render, screen } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import PropTypes from 'prop-types';
import ListTakenTimes from '../ListTakenTimes';

const MockChildren = ({ takenTime }) => (
  <div>{takenTime.name}</div>
);

MockChildren.propTypes = {
  takenTime: PropTypes.shape({ id: PropTypes.number, name: PropTypes.string }).isRequired,
};

const takenTime = {};
const renderReadyComponent = takenTimes => (
  <ListTakenTimes takenTimes={takenTimes}>
    <MockChildren takenTime={takenTime} />
  </ListTakenTimes>
);
beforeEach(() => {

});

describe('<ListTakenTimes />', () => {
  describe('If takenTimes is not empty', () => {
    const takenTimes = [
      { id: 1, name: 'First Taken Time' },
      { id: 2, name: 'Second Taken Time' },
    ];
    it('renders children with taken times', () => {
      render(renderReadyComponent(takenTimes));

      expect(screen.getByText(/First Taken Time/i)).toBeInTheDocument();
      expect(screen.getByText(/Second Taken Time/i)).toBeInTheDocument();
    });
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent(takenTimes));
      expect(renderedContainer).toMatchSnapshot();
    });
  });

  describe('If takenTimes is not empty', () => {
    const takenTimes = [];
    it('shows a text ', () => {
      render(renderReadyComponent(takenTimes));
      expect(screen.getByText(/There is no anything to show/i)).toBeInTheDocument();
    });
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent(takenTimes));
      expect(renderedContainer).toMatchSnapshot();
    });
  });
});
