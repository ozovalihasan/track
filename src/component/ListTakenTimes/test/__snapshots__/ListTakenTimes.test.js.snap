// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListTakenTimes /> If takenTimes is not empty renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ kOhOls"
      >
        <div>
          First Taken Time
        </div>
        <div>
          Second Taken Time
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ kOhOls"
    >
      <div>
        First Taken Time
      </div>
      <div>
        Second Taken Time
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ListTakenTimes /> If takenTimes is not empty renders correctly 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ kOhOls"
      >
        There is no anything to show
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ kOhOls"
    >
      There is no anything to show
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
